### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by aboongm**
- Use jest framework for testing.
- Create utilityFunctions.test.js for testing the add item and delete item functions to test.
- Mock a storage object to "imitate" localStorage operations 
- Mock HTML to test if add/delete functions to add or remove
- Group the tests using the describe() method.

### CODE REVIEW

File: src/module/utilityFunctions.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import Task from './Task.js';
-import * as Elements from './constElements.js';
+import { taskInput, taskList } from './constElements.js';
 
 const Tasks = new Task();
 
 const displayContent = () => {
-  Elements.taskList.innerHTML = '';
+  // taskList.innerHTML = '';
```
Comment (brytebee) [inline src/module/utilityFunctions.js]: - [ ] **Required:** Kindly remove commented code. It is unprofessional to commit commented code.

File: src/module/utilityFunctions.js [hunk 2]
```diff
@@ -19,18 +19,21 @@ const displayContent = () => {
             </div>
       `;
   });
+  // window.location.reload();
```
Comment (brytebee) [inline src/module/utilityFunctions.js]: - [ ] **Required:** Kindly remove commented code.

File: src/module/utilityFunctions.js [hunk 3]
```diff
@@ -19,18 +19,21 @@ const displayContent = () => {
...
 };
 
 const checkLocalStorage = () => {
   if (JSON.parse(localStorage.getItem('TASKS_LIST')) != null) {
     Tasks.TaskObject = JSON.parse(localStorage.getItem('TASKS_LIST'));
-    displayContent();
+    // displayContent();
```
Comment (brytebee) [inline src/module/utilityFunctions.js]: - [ ] **Required:** Kindly remove commented code.

File: src/module/utilityFunctions.js [hunk 4]
```diff
@@ -19,18 +19,21 @@ const displayContent = () => {
...
   }
 };
 
 const addTask = () => {
-  Tasks.add(Elements.taskInput);
-  checkLocalStorage();
+  Tasks.add(taskInput);
+  // checkLocalStorage();
```
Comment (brytebee) [inline src/module/utilityFunctions.js]: - [ ] **Required:** Kindly remove commented code.

File: src/module/utilityFunctions.js [hunk 5]
```diff
@@ -46,7 +49,8 @@ const removeTask = (element) => {
     Tasks.remove(i);
     reorderTaskObjectId(Tasks.TaskObject);
     localStorage.setItem('TASKS_LIST', JSON.stringify(Tasks.TaskObject));
-    checkLocalStorage();
+    // displayContent();
```
Comment (brytebee) [inline src/module/utilityFunctions.js]: - [ ] **Required:** Kindly remove commented code.

File: src/index.js [hunk 6]
```diff
@@ -3,36 +3,43 @@ import {
...
   Tasks.TaskObject = [];
   localStorage.setItem('TASKS_LIST', JSON.stringify(Tasks.TaskObject));
-  checkLocalStorage();
+  displayContent();
 });
 
 // tasklist functionalities
-Elements.taskList.addEventListener('click', (e) => {
+taskList.addEventListener('click', (e) => {
   // e.stopPropagation();
```
Comment (brytebee) [inline src/index.js:40]: - [ ] **Required:** Kindly remove commented code.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi Team👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- good commit messages✔️
- no console errors✔️
- no linter errors✔️
- nicely mocked the local storage✔️
- nicely mocked the DOM✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShoiraTa** [comment]
## **STATUS: APPROVED  🟢 ** 
------
### Hi team, 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**